### PR TITLE
add test for a NULL local work size and a required work-group size

### PR DESCRIPTION
**Pull request by bashbaug**
see #2501 

This tests the following scenarios:

1. Execute a kernel with a required work-group size, passing `NULL` as the local work size.
2. Query the suggested work-group size for a kernel with a required work-group size.

### CODE REVIEW

File: test_conformance/api/test_kernel_attributes.cpp [hunk 1]
```diff
@@ -336,3 +336,160 @@ REGISTER_TEST(kernel_attributes)
     }
     return success ? TEST_PASS : TEST_FAIL;
 }
+
+REGISTER_TEST(null_required_work_group_size)
+{
+    cl_int error = CL_SUCCESS;
+
+    bool checkSuggestedLocalWorkSize = false;
```
Comment (EwanC) [inline test_conformance/api/test_kernel_attributes.cpp]: Bit of a nitpick, but think the surrounding variable naming style tends to snake_case (same comment for a couple other variables later on). Also not sure if we need a specific variable for this, or can  just check  `clGetKernelSuggestedLocalWorkSizeKHR` isn't `nullptr` later on.

Comment (bashbaug) [reply test_conformance/api/test_kernel_attributes.cpp]: Hmm, good point.  I'll just check if the function pointer is `nullptr`.

I'll see what I can do about snake_case vs. camelCase also.  The CTS seems to use both, but snake_case is definitely more pervasive in this file.  Thanks!

**Review by EwanC — DISMISSED** [review]
